src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/wrapper/PKCS11Exception.java

```diff
@@ -63,229 +63,160 @@
+        register("CKR_TEMPLATE_INCONSISTENT", 0x000000D1L);
+        register("CKR_TOKEN_NOT_PRESENT", 0x000000E0L);
+        register("CKR_TOKEN_NOT_RECOGNIZED", 0x000000E1L);
+        register("CKR_TOKEN_WRITE_PROTECTED", 0x000000E2L);
+        register("CKR_UNWRAPPING_KEY_HANDLE_INVALID", 0x000000F0L);
+        register("CKR_UNWRAPPING_KEY_SIZE_RANGE", 0x000000F1L);
+        register("CKR_UNWRAPPING_KEY_TYPE_INCONSISTENT", 0x000000F2L);
+        register("CKR_USER_ALREADY_LOGGED_IN", CKR_USER_ALREADY_LOGGED_IN);
+        register("CKR_USER_NOT_LOGGED_IN", CKR_USER_NOT_LOGGED_IN);
+        register("CKR_USER_PIN_NOT_INITIALIZED", 0x00000102L);
+        register("CKR_USER_TYPE_INVALID", 0x00000103L);
+        register("CKR_USER_ANOTHER_ALREADY_LOGGED_IN", 0x00000104L);
+        register("CKR_USER_TOO_MANY_TYPES", 0x00000105L);
+        register("CKR_WRAPPED_KEY_INVALID", 0x00000110L);
+        register("CKR_WRAPPED_KEY_LEN_RANGE", 0x00000112L);
+        register("CKR_WRAPPING_KEY_HANDLE_INVALID", 0x00000113L);
+        register("CKR_WRAPPING_KEY_SIZE_RANGE", 0x00000114L);
+        register("CKR_WRAPPING_KEY_TYPE_INCONSISTENT", 0x00000115L);
+        register("CKR_RANDOM_SEED_NOT_SUPPORTED", 0x00000120L);
+        register("CKR_RANDOM_NO_RNG", 0x00000121L);
+        register("CKR_DOMAIN_PARAMS_INVALID", 0x00000130L);
+        register("CKR_CURVE_NOT_SUPPORTED", 0x00000140L);
+        register("CKR_BUFFER_TOO_SMALL", CKR_BUFFER_TOO_SMALL);
+        register("CKR_SAVED_STATE_INVALID", 0x00000160L);
+        register("CKR_INFORMATION_SENSITIVE", 0x00000170L);
+        register("CKR_STATE_UNSAVEABLE", 0x00000180L);
+        register("CKR_CRYPTOKI_NOT_INITIALIZED", 0x00000190L);
+        register("CKR_CRYPTOKI_ALREADY_INITIALIZED",
+                CKR_CRYPTOKI_ALREADY_INITIALIZED);
+        register("CKR_MUTEX_BAD", 0x000001A0L);
+        register("CKR_MUTEX_NOT_LOCKED", 0x000001A1L);
+        register("CKR_NEW_PIN_MODE", 0x000001B0L);
+        register("CKR_NEXT_OTP", 0x000001B1L);
+        register("CKR_EXCEEDED_MAX_ITERATIONS", 0x000001B5L);
+        register("CKR_FIPS_SELF_TEST_FAILED", 0x000001B6L);
+        register("CKR_LIBRARY_LOAD_FAILED", 0x000001B7L);
+        register("CKR_PIN_TOO_WEAK", 0x000001B8L);
+        register("CKR_PUBLIC_KEY_INVALID", 0x000001B9L);
+        register("CKR_FUNCTION_REJECTED", 0x00000200L);
+        register("CKR_TOKEN_RESOURCE_EXCEEDED", 0x00000201L);
+        register("CKR_OPERATION_CANCEL_FAILED", 0x00000202L);
+        register("CKR_VENDOR_DEFINED", 0x80000000L);
+    };
+
+    private static void register(String name, long errorCode) {
+        errorMap.put(errorCode, name);
     }
 
+    private static String lookup(long errorCode) {
+        String res = errorMap.get(errorCode);
+        // for unknown PKCS11 return values, just use hex as its string
+        if (res == null) {
+            res = "0x" + Functions.toFullHexString((int)errorCode);
+            errorMap.put(errorCode, res);
```

Comment:
It would be nice if the errorMap is immutable once the known error codes get registered.